storage-proofs/porep/src/stacked/vanilla/proof.rs

```diff
@@ -286,20 +292,144 @@ impl<'a, Tree: 'static + MerkleTreeTrait, G: 'static + Hasher> StackedDrg<'a, Tr
         Ok(())
     }
 
-    #[allow(clippy::type_complexity)]
-    fn generate_labels(
+    fn prepare_layers(
+        graph: &StackedBucketGraph<Tree::Hasher>,
+        config: &StoreConfig,
+        layers: usize,
+    ) -> Vec<LayerState> {
+        let label_configs = (1..=layers).map(|layer| {
+            StoreConfig::from_config(&config, CacheKey::label_layer(layer), Some(graph.size()))
+        });
+
+        let mut states = Vec::with_capacity(layers);
+        for (layer, label_config) in (1..=layers).zip(label_configs) {
+            // Check if this layer is already on disk
+            let data_path = StoreConfig::data_path(&label_config.path, &label_config.id);
+            let generated = data_path.exists()
+                && DiskStore::<<Tree::Hasher as Hasher>::Domain>::new_from_disk(
+                    graph.size(),
+                    Tree::Arity::to_usize(),
+                    &label_config,
+                )
+                .is_ok();
+            if generated {
+                // successfull load
+                info!("found valid labels for layer {}", layer);
+            }
+
+            states.push(LayerState {
+                config: label_config,
+                generated,
+            });
+        }
+
+        states
+    }
+
+    fn generate_labels_replicate(
```

Comment:
Can you come up with a more descriptive name and/or add a comment to explain exactly what this does? We're getting into a soup of self-referential names which is hard to get a handle on by just looking at one part. It would be good to reverse that trend where we can.

Reply:
added some docs and improved the name a little, honestly can't come up with something better